test/apps/iodemo/io_demo.cc

```diff
@@ -816,18 +816,74 @@ class DemoServer : public P2pDemoCommon {
         MemoryPool<IoWriteResponseCallback>& _pool;
     };
 
-    typedef struct {
-        long    read_count;
-        long    write_count;
-        long    active_conns;
-    } state_t;
+    class ConnectionStat {
+    public:
+        ConnectionStat() {
+            reset();
+        }
+
+        void reset() {
+            memset(_bytes_counters, 0, sizeof(_bytes_counters));
+            memset(_op_counters, 0, sizeof(_op_counters));
+        }
+
+        template<io_op_t op_type> long*
+        get_completion_counter() {
+            UCS_STATIC_ASSERT(op_type < IO_OP_MAX);
+            return &_op_counters[op_type];
+        }
+
+        template<io_op_t op_type> long
+        get_completion_count() {
+            return *get_completion_counter<op_type>();
+        }
+
+        template<io_op_t op_type> long
+        get_bytes() {
+            return *get_bytes_counter<op_type>();
+        }
+
+        template<io_op_t op_type> void
+        op_started(size_t bytes) {
+            *get_bytes_counter<op_type>() += bytes;
```

Comment:
remove these wrappers and just keep `long& bytes<op>()` and `long& completions<op>()` (make them public)?